cpmpy/expressions/core.py

```diff
@@ -401,6 +401,11 @@ def __init__(self, name, arg_list):
         # sanity checks
         assert (name in Operator.allowed), "Operator {} not allowed".format(name)
         arity, is_bool = Operator.allowed[name]
+        if is_bool:
```

Comment:
I'm not sure I understand, just remove the flatlist?
the code is only checking the direct children, the flatlist call just handles cases like wsum were the args consist of 2 list, in stead of one. It can be removed however since there are no such cases for boolean operators